File: dropwizard/src/test/java/com/yammer/flopwizard/example/HelloWorldResource.java
package com.yammer.flopwizard.example;

import javax.ws.rs.*;
import javax.ws.rs.core.MediaType;
import javax.ws.rs.core.Response;
import javax.ws.rs.core.Response.Status;
import java.util.Collections;
import java.util.List;
import java.util.concurrent.atomic.AtomicInteger;

@Path("/hello-world")
@Produces(MediaType.APPLICATION_JSON)
public class HelloWorldResource {
    private final AtomicInteger id;
    private final String saying;

    public HelloWorldResource(String saying) {
        this.id = new AtomicInteger();
        this.saying = saying;
    }

    @GET
    public List<Saying> sayHello() {
        return Collections.singletonList(new Saying(id.incrementAndGet(), saying));
    }

    @POST
    public Response intentionalError() {
        return Response.status(Status.BAD_REQUEST).build();
    }

    @PUT
    public void unintentionalError() {
        final String blah = null;
        blah.charAt(0);
    }
}

File: dropwizard_2.9.1/src/test/scala/com/yammer/dropwizard/examples/HelloWorldResource.scala
package com.yammer.dropwizard.examples

import javax.ws.rs._
import core.Response.Status
import core.{Response, MediaType}

@Path("/hello-world")
@Produces(Array(MediaType.APPLICATION_JSON))
class HelloWorldResource(saying: String) {
  @GET
  def sayHello(@QueryParam("opt") opt: Option[String]) = Seq(saying)

  @POST
  def intentionalError = Response.status(Status.BAD_REQUEST).build()

  @PUT
  def unintentionalError = None.get
}
